lib/pbench/server/filetree.py

```diff
@@ -339,68 +359,51 @@ def unpack(self, incoming: Path, results: Path):
+            try:
+                process = self.subprocess_run(tar_command, unpacked)
+            except Exception as exc:
+                raise TarballUnpackError(self.tarball_path, str(exc)) from exc
+            else:
+                if process.returncode != 0:
+                    raise TarballUnpackError(
+                        self.tarball_path,
+                        f"tar exited with status {process.returncode}:  {process.stderr.strip()!r}",
+                    )
+
+            find_command = "find . ( -type d -exec chmod ugo+rx {} + ) -o ( -type f -exec chmod ugo+r {} + )"
+            try:
+                process = self.subprocess_run(find_command, unpacked)
+            except Exception as exc:
+                raise TarballModeChangeError(unpacked, str(exc)) from exc
+            else:
+                if process.returncode != 0:
+                    raise TarballModeChangeError(
+                        unpacked,
+                        f"tar exited with status {process.returncode}:  {process.stderr.strip()!r}",
+                    )
+
+            try:
+                shutil.move(unpacked / self.name, incoming)
+            except Exception as exc:
+                try:
+                    shutil.rmtree(incoming / self.name)
+                except Exception as e:
+                    self.logger.error(
+                        f"Error in removing partially copied Tarball: {str(e)}"
+                    )
                 raise TarballUnpackError(
                     self.tarball_path,
-                    f"tar exited with status {process.returncode}:  {process.stderr.strip()!r}",
+                    f"Error moving {str(unpacked / self.name)!r} to {str(incoming)!r}: {str(exc)}",
                 )
-
-        find_command = "find . ( -type d -exec chmod ugo+rx '{}' + ) -o ( -type f -exec chmod ugo+r '{}' + )"
-        try:
-            process = subprocess.run(
-                find_command.split(),
-                cwd=unpacked,
-                stdin=subprocess.DEVNULL,
-                capture_output=True,
-                text=True,
-            )
-        except Exception as exc:
-            raise TarballModeChangeError(unpacked, str(exc)) from exc
-        else:
-            if process.returncode != 0:
-                raise TarballModeChangeError(
-                    unpacked,
-                    f"tar exited with status {process.returncode}:  {process.stderr.strip()!r}",
-                )
-
-        try:
-            shutil.move(unpacked / self.name, incoming)
-        except Exception as exc:
+        finally:
             try:
-                shutil.rmtree(incoming)
-            except Exception as e:
+                shutil.rmtree(unpacked, ignore_errors=True)
+            except Exception as exc:
                 raise TarballUnpackError(
-                    incoming, f"Error in removing partially copied Tarball: {str(e)}"
+                    unpacked,
```

Comment:
If we're here due to a previous failure, then I don't think we want to raise a new exception if the `rmtree()` fails (the `finally` block will re-raise the current exception when the block is exited).  If we're here _without_ incurring an exception, then this function has successfully unpacked the tarball, so I don't think we want to raise an exception in that case either (or, if we really do want to raise an exception in that case, then we should be removing the `incoming / self.name` tree, first!).

Nevertheless, given that we've specified `ignore_errors=True`, I don't think we're ever going to see a failure here, so this is all moot...except for the fact that the `try` block is pointless.  So, I suggest removing it and just calling `rmtree()`.